flytekit/core/workflow.py

```diff
@@ -634,6 +634,40 @@ def add_launch_plan(self, launch_plan: _annotated_launch_plan.LaunchPlan, **kwar
     def add_subwf(self, sub_wf: WorkflowBase, **kwargs) -> Node:
         return self.add_entity(sub_wf, **kwargs)
 
+    def add_on_failure_handler(self, entity):
```

Comment:
more of a question (it's fine if the answer is no): does it make sense to make this specific to eager tasks?

Reply:
i think it's okay actually.  it should work generally and kinda closes this out too https://github.com/flyteorg/flyte/issues/5913 which pingsutw linked me to.  Though I didn't do what that ticket suggested.  I didn't actually remember that ticket at all, and had it in the constructor to begin with, but it made more sense as a separate call when implementing it, but happy to revisit.